app.js

```diff
@@ -1,8 +1,74 @@
+// Convert text to only alphabetic characters and spaces
+const removePunctuation = text => {
+    return text.replace(/[^ A-Za-z]/g,'');
+};
+
+// Convert string of text into an array of individual words
+const splitText = text => {
+    return text.split(' ');
+}
+
+// Remove banned words from an array of words
+const removeBannedWords = (words, banned) => {
+    const bannedWords = new Set(banned);
+    return words.filter(word => !bannedWords.has(word));
+}
+
+// Count the frequency of each word in an array of words
+const getWordCounts = words => {
+    const wordCounts = {};
+
+    for (let word of words) {
+        if (typeof wordCounts[word] === 'number') {
+            wordCounts[word]++;
+        } else {
+            wordCounts[word] = 1;
+        }
+    }
+
+    return wordCounts;
+}
+
+// Given an array of words with their respective frequencies, return
+// the word with the highest frequency
+const getCommonWord = wordCounts => {
+    let highestFrequency = 0;
+    let commonWord = '';
+    
+    for (let [word, count] of Object.entries(wordCounts)) {
+        if (count > highestFrequency) {
+            highestFrequency = count;
+            commonWord = word;
+        }
+    }
+
+    return commonWord;
+} 
+
 /**
  * @param {string} paragraph
  * @param {string[]} banned
  * @return {string}
  */
 const mostCommonWord = (paragraph, banned) => {
+    // Remove punctuation from paragraph and convert to lowercase
+    const text = removePunctuation(paragraph).toLowerCase();
+
+    // Split text into individual words
+    words = splitText(text);
+
+    // Remove banned words from text
```

Comment:
Your function naming is excellent! Almost to the point you could leave off these comments in mostCommonWord. I'd almost rather see more comments in the helper functions (specifically getCommonWord), even if it means summarizing a bit of basic JS methods (like Object.entries)